### PR TITLE
Import Robolectric Native Graphics tests

### PR DESCRIPTION
**Pull request by hoisie**
Import Robolectric Native Graphics tests

Create a new integration_test project, 'nativegraphics', to contain these tests.
The advantage of an integration test project is that it can use AGP and have
access to resources.

### CODE REVIEW

**Comment by utzcoz** [comment]
@hoisie Looks like https://android-review.googlesource.com/c/platform/external/robolectric/+/2492289 AOSP wants to merge this PR, what about merging it first?

**Comment by ihcinihsdk** [comment]
> 

@hoisie @utzcoz Hi Michael and utzcoz,
I was creating https://android-review.googlesource.com/c/platform/external/robolectric/+/2492289 to work on enable the test, and was planning to wait for the merge of this PR and grab the code via `git fetch / pull` etc.

Having said that, do you have an ETA for merging this PR?

**Comment by hoisie** [comment]
Hi @ihcinihsdk working on this now :)

**Comment by ihcinihsdk** [comment]
> Hi @ihcinihsdk working on this now :)

Thanks a lot, Michael!

**Comment by utzcoz** [comment]
@hoisie Does RNG support webp? I remember there is a webp supporting problem of previous resource mechanism.

**Comment by hoisie** [comment]
@utzcoz yup it supports WebP

File: .github/workflows/tests.yml
Comment (utzcoz) [inline .github/workflows/tests.yml:68]: @hoisie Is it for test purpose only?

Comment (hoisie) [reply .github/workflows/tests.yml:68]: It's complicated... I was seeing weird CI failures when running native graphics tests in the regular tests CI workflow. If I separate the graphics tests so they are in a separate process it fixes the issue.

I am thinking of running the graphics tests on our M1 mini runner for now.

We may also upgrade to a paid plan to use larger CI machines with more than 7 GiB ram.

Comment (utzcoz) [reply .github/workflows/tests.yml:68]: Oh. Looks like all tests consume too much RAM. It's great to separate native graphics tests.

> We may also upgrade to a paid plan to use larger CI machines with more than 7 GiB ram.

Is it necessary? At least current CI time is accepted after nativeruntime building removed. We can trade time for space now.

**Review by utzcoz — APPROVED** [review]

**Comment by utzcoz** [comment]
> @utzcoz yup it supports WebP

Cool. The RNG is awesome. Looks like this PR can be merged because CI passed.

**Comment by hoisie** [comment]
@utzcoz I'll merge it via copybara, needs to get a couple other approvals first.

Yeah it does seem like Error 137 comes is triggered by the GitHub actions host machine when it kills the process due to exceeding memory thresholds:

https://stackoverflow.com/questions/68844666/github-action-is-being-killed

We have been occasionally seeing this error 137 in other places, not just RNG tests. 

Robolectric's test suite is very large and complex, it's pretty amazing it that it has been running in under 7 GiB of RAM for so long.

I think the plan is to experiment with larger machines for a while, and if the cost is reasonable, we can go with the ones with 16 GiB ram.

**Pull request closed without merging**
